mne/tests/test_epochs.py

```diff
@@ -2450,6 +2450,48 @@ def test_events_list():
     assert_array_equal(epochs.events, np.array(events))
 
 
+def test_save_overwrite(tmpdir):
+    """Test saving with overwrite functionality."""
+
+    tempdir = str(tmpdir)
+    raw = mne.io.RawArray(np.random.RandomState(0).randn(100, 10000),
+                          mne.create_info(100, 1000.))
+
+    events = mne.make_fixed_length_events(raw, 1)
+    epochs = mne.Epochs(raw, events)
+
+    # scenario 1: overwrite=False (default) and there isn't a file to overwrite
+    # make a filename that has not already been saved to
+    fname1 = op.join(tempdir, 'test' + str(np.random.rand()) + '-epo.fif')
+    while op.isfile(fname1):
+        fname1 = op.join(tempdir, 'test' + str(np.random.rand()) + '-epo.fif')
+    # run function to be sure it doesn't throw an error
+    epochs.save(fname1)
+    # check that the file got written
+    assert(op.isfile(fname1))
+
+    # scenario 2: overwrite=False (default) and there is a file to overwrite
+    # fname1 exists because of scenario 1 above
+    pytest.raises(OSError, epochs.save, fname1)
+
+    # scenario 3: overwrite=True and there isn't a file to overwrite
+    # make up a filename that has not already been saved to
+    fname2 = op.join(tempdir, 'test' + str(np.random.rand()) + '-epo.fif')
+    while op.isfile(fname2):
```

Comment:
debugging artifact? I can't seem to make sense of this while loop ...

Reply:
I was overdoing it: The while loop makes sure that fname2 really doesn't exist by generating a new file name until it doesn't find that filename already saved. It seems like it's not necessary though, so I removed it.